coordinatorlib/src/main/scala/com/socrata/datacoordinator/truth/loader/sql/SqlLoader.scala

```diff
@@ -197,15 +224,38 @@ final class SqlLoader[CT, CV](val connection: Connection,
       case None => None
     }
 
-  def upsert(jobId: Int, row: Row[CV]): Unit = {
+  def upsert(jobId: Int, row: Row[CV], bySystemId: Boolean): Unit = {
     checkJob(jobId)
-    idOf(row) match {
-      case Some(id) =>
-        if(currentBatch.hasUpsertFor(id)) {
+    // okay, there are three cases here:
+    // isSystemPK and bySystemId is true or false:
+    //   - use system id for updates
+    //   - system id not required for insert
+    // isUserPK and bySystemId is false:
+    //   - user user pk for updates and inserts
+    // isUserPK and bySystemId is true:
+    //   - allow for system id to be used on update
+    //     if user pk is not passed (prefer user pk if given)
```

Comment:
So, we don't allow updating the user primary key, but I'm pretty confident that if you chose to do updates here by the system id and allowed a user primary key, that would support exactly that.

So... just ignoring the system id in the case the user primary key is already given seems like the simplest behavior to choose, given that is already what we do when we have a user primary key.

Reply:
Ok, as long as we don't support updating the user primary key then seems reasonable.

If we wanted to support that in the future, then seems modifying this would be an easy way to let us do that.

Reply:
👍 agreed